marimo/_save/stores/store.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2024 Marimo. All rights reserved.
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING
+
+from marimo._save.cache import (
+    Cache,
+)
+
+if TYPE_CHECKING:
+    from marimo._save.hash import HashKey
+    from marimo._save.loaders import BasePersistenceLoader as Loader
+
+
+class Store(ABC):
+    _instance = None
+
+    # Singleton
+    def __new__(cls) -> Store:
+        if cls._instance is None:
+            cls._instance = super().__new__(cls)
+            # Put any initialization here.
+        return cls._instance
+
+    @abstractmethod
+    def get(self, key: HashKey, loader: Loader) -> bytes:
```

Comment:
i wonder if `get` should return `Optional[bytes]`. so you don't need to check `hit` before which could be two networks requests instead of one. although hit is likely cheap, it's still a waterfall.

`hit` should still exist for the case where you only need to check existence